Add a topic guide for response function vocabulary (#111)
sunpy/sunkit-instruments: docs/topic_guide/channel_response.rst

.. _sunkit-instruments-topic-guide-channel-response:

************************************
A Vocabulary for Instrument Response
************************************

> This topic guide provides a vocabulary for response functions for narrow-band imaging instruments.

Reviewer: ```suggestion
This topic guide provides a vocabulary for response functions for imaging instruments.
```

The goal of providing a single vocabulary for instrument response calculations is to provide a specification for a common interface that can be used by multiple instruments.
This reduces the amount of effort needed to develop analysis software for new instruments and enables cross-instrument comparisons as upstream packages and users can program against a single interface for these response calculations.
An abstract implementation of this vocabulary is provided in this package.

Temperature Response
--------------------

The temperature response describes the instrument sensitivity as a function of temperature.
It is a useful quantity when performing thermal analysis of imaging data, such as a differential emission measure or filter ratio analysis.
The temperature response is defined as,

.. math::

    K(T) = \int\mathrm{d}\lambda\,R(\lambda)S(\lambda,T)\quad[\mathrm{DN}\,\mathrm{pixel}^{-1}\,\mathrm{s}^{-1} \,\mathrm{cm}^5]

> It has a physical type of DN per pixel per unit time per unit emission measure.

Author: Fair point, especially since the astropy unit definition is not overly descriptive: https://docs.astropy.org/en/stable/units/index.html#module-astropy.units.astrophys.

The temperature response is related to the observed intensity in a given pixel by,

.. math::

    I = \int\mathrm{d}T\,K(T)\mathrm{EM}(T)\quad[\mathrm{DN}\,\mathrm{pixel}^{-1}\,\mathrm{s}^{-1}],

where :math:`\mathrm{EM}(T)` is the line-of-sight emission measure distribution in a given pixel.

Source Spectra
--------------

The source spectra describes how a source is emitting as a function of wavelength and temperature.
It is denoted by :math:`S(\lambda, T)`.
The source spectra has a physical type of photon per unit time per unit wavelength per solid angle per unit density.
The units are commonly expressed as
:math:`\mathrm{photon}\,\mathrm{s}^{-1}\,\mathring{\mathrm{A}}^{-1}\,\mathrm{sr}^{-1}\,\mathrm{cm}^3`.
The source spectra is specified by the user and can be computed from atomic databases (e.g. CHIANTI).
This quantity is independent of any instrument properties.

Wavelength Response
-------------------

The wavelength response describes the instrument sensitivity as a function of wavelength and time.
The wavelength response is defined as,

.. math::

>     R(\lambda,t) = A_{\mathrm{eff}}(\lambda,t)f(\lambda)\frac{p}{gs}\quad[\mathrm{cm}^2\,\mathrm{DN}\,\mathrm{sr}\,\mathrm{photon}^{-1}\,\mathrm{pixel}^{-1}]

Author: ```suggestion
    R(\lambda,t) = A_{\mathrm{eff}}(\lambda,t)f(\lambda)\frac{pg}{s}\quad[\mathrm{cm}^2\,\mathrm{DN}\,\mathrm{sr}\,\mathrm{photon}^{-1}\,\mathrm{pixel}^{-1}]
```

It has a physical type of area DN solid angle per photon per pixel.

Camera Gain
-----------

The camera gain, :math:`g`, describes the conversion between electrons and data number (DN).
This is a property of the detector.

> The units of the camera gain are as :math:`\mathrm{electron}\,\mathrm{DN}^{-1}`.

Author: ```suggestion
The units of the camera gain are as :math:`\mathrm{DN}\,\mathrm{electron}^{-1}`.
```

Author: I believe gain is typically measured in terms of how many DN are produced by 1 elecrron

Photon-to-Energy Conversion

> ---------------------------

Author: Yes it does! In fact, this is how I've implemented it in #98.

The photon-to-energy conversion is given by the amount of energy carried by a photon of wavelength :math:`\lambda`,

.. math::

    f(\lambda) = \frac{hc}{\lambda}\quad[\mathrm{eV}\,\mathrm{photon}^{-1}]

> where :math:`h` is Planck's constant and :math:`c` is the speed of light.
> It has a physical type of energy per photon.

Reviewer 2: ```suggestion
where :math:`h` is Planck's constant and :math:`c` is the speed of light.
It has a physical type of energy per photon.

Use `~astropy.units.spectral` to provide a list of appropriate `astropy.units` equivalencies for this conversion.
```
Feel free to adjust!  This also assumes intersphinx is set up.

Energy-to-Electron Conversion
-----------------------------

The energy-to-electron conversion, :math:`s`, describes the conversion between electrons released in the detector and the energy of an incoming photon.
It has a physical type of energy per electron.
For silicon detectors, a value of :math:`s=3.65\,\mathrm{eV}\,\mathrm{electron}^{-1}` is typically used as this is approximately the energy required to free an electron in silicon.

> Pixel Area

Author: Upon further reflection, I think this needs a more specific name, either "Pixel Solid Angle" or "Pixel Angular Area". "Pixel Area" potentially conflicts with the *physical* area of the pixel, i.e. something with units of cm$^2$ per pixel.

.. _sunkit-instruments-topic-guide-channel-response:

************************************
A Vocabulary for Instrument Response
************************************

> This topic guide provides a vocabulary for response functions for imaging instruments.
> The goal of providing a single vocabulary for instrument response calculations is to provide a specification for a common interface that can be used by multiple instruments.
> This reduces the amount of effort needed to develop analysis software for new instruments and enables cross-instrument comparisons as upstream packages and users can program against a single interface for these response calculations.
> An abstract implementation of this vocabulary is provided in this package.

Reviewer 3: ```suggestion
This topic guide provides a vocabulary for response functions for imaging instruments.

The reason to provide a single vocabulary for instrument response calculations is to define a specification for a common interface that can be used by multiple instruments.
This reduces the amount of effort needed to develop analysis software for new instruments and enables cross-instrument comparisons as upstream packages and users can program against a single interface for these response calculations.

An abstract implementation of this vocabulary is provided in this package.
```

Author: What's wrong with one paragraph?

Reviewer 3: There are some text changes mixed in there. Let me make it one paragraph

Temperature Response
--------------------

The temperature response describes the instrument sensitivity as a function of temperature.
It is a useful quantity when performing thermal analysis of imaging data, such as a differential emission measure or filter ratio analysis.
The temperature response is defined as,

.. math::

    K(T) = \int\mathrm{d}\lambda\,R(\lambda)S(\lambda,T)\quad[\mathrm{DN}\,\mathrm{pixel}^{-1}\,\mathrm{s}^{-1} \,\mathrm{cm}^5]

It has a physical type of data number (DN) per pixel per unit time per unit emission measure.
Note that the temperature response is a function of *both* the instrument properties as well as the atomic physics of the emitting source.
The temperature response is related to the observed intensity in a given pixel by,

.. math::

>     I = \int\mathrm{d}T\,K(T)\mathrm{EM}(T)\quad[\mathrm{DN}\,\mathrm{pixel}^{-1}\,\mathrm{s}^{-1}],

Author: I just realized this is actually wrong. The integral is taken over the *differential* emission measure distribution. 
```suggestion
    I = \int\mathrm{d}T\,K(T)\mathrm{DEM}(T)\quad[\mathrm{DN}\,\mathrm{pixel}^{-1}\,\mathrm{s}^{-1}],
```

> where :math:`\mathrm{EM}(T)` is the line-of-sight emission measure distribution in a given pixel.

Author: ```suggestion
where :math:`\mathrm{DEM}(T)=n^2\frac{dh}{dT}` is the line-of-sight *differential* emission measure distribution in a given pixel and is typically express in units of :math:`\mathrm{cm}^{-5}\,\mathrm{K}^{-1}`. 
```

Source Spectra
--------------

> The source spectra describes how a source is emitting as a function of wavelength and temperature.

Reviewer 4: This is close to the spectral radiance (as defined by wikipedia)

Author: It's close modulo a factor of cm$^{-5}$, i.e. the emission measure. I'm hesitant to call it the spectral radiance since the product of this and the emission measure would actually be the spectral radiance in the appropriate units.

Author: I've added a note in the description that this may also be referred to as the spectral radiance per unit emission measure.

Reviewer 4: Ah yes, I was reading too quickly. Spectral radiance per unit emission measure sounds good to me.

It is denoted by :math:`S(\lambda, T)`.
The source spectra has a physical type of photon per unit time per unit wavelength per solid angle per unit density.
The units are commonly expressed as
:math:`\mathrm{photon}\,\mathrm{s}^{-1}\,\mathring{\mathrm{A}}^{-1}\,\mathrm{sr}^{-1}\,\mathrm{cm}^3`.
The source spectra is specified by the user and can be computed from atomic databases (e.g. CHIANTI).
This quantity is independent of any instrument properties.

Wavelength Response
-------------------

The wavelength response describes the instrument sensitivity as a function of wavelength and time.
The wavelength response is defined as,

.. math::

    R(\lambda,t) = A_{\mathrm{eff}}(\lambda,t)f(\lambda)\frac{pg}{s}\quad[\mathrm{cm}^2\,\mathrm{DN}\,\mathrm{sr}\,\mathrm{photon}^{-1}\,\mathrm{pixel}^{-1}]

> It has a physical type of area DN solid angle per photon per pixel.

Reviewer 3: I don't understand the "area DN solid angle" part of this unit.

Author: Yeah I was just looking at this. I think it's the ordering that makes things confusing. It should be area DN per photon solid angle per pixel. Same with the units in the equation.

Camera Gain
-----------

The camera gain, :math:`g`, describes the conversion between electrons and data number (DN).
This is a property of the detector.
The units of the camera gain are as :math:`\mathrm{DN}\,\mathrm{electron}^{-1}`.

Photon-to-Energy Conversion
---------------------------

The photon-to-energy conversion is given by the amount of energy carried by a photon of wavelength :math:`\lambda`,

.. math::

    f(\lambda) = \frac{hc}{\lambda}\quad[\mathrm{eV}\,\mathrm{photon}^{-1}]

where :math:`h` is Planck's constant and :math:`c` is the speed of light.
It has a physical type of energy per photon.

Use `~astropy.units.spectral` to provide a list of appropriate `astropy.units` equivalencies for this conversion.

Energy-to-Electron Conversion
-----------------------------

> The energy-to-electron conversion, :math:`s`, describes the conversion between electrons released in the detector and the energy of an incoming photon.

Reviewer 4: This is often called electron-hole-pair-creation energy in the literature, but that might be overly explicit for our audience.

Author: I can add a note in the description to this effect.

It has a physical type of energy per electron.
For silicon detectors, a value of :math:`s=3.65\,\mathrm{eV}\,\mathrm{electron}^{-1}` is typically used as this is approximately the energy required to free an electron in silicon.

Pixel Solid Angle
-----------------

The pixel area, :math:`p`, is the angular area in the plane of the sky subtended by a single detector pixel.
It has a physical type of solid angle per pixel.
The units of the pixel area are typically expressed as :math:`\mathrm{sr}\,\mathrm{pixel}^{-1}`.
Typically, this quantity can be determined as the product the spatial plate scale in each direction.

> In the FITS standard, these keys are denoted by `CDELTi`, with `i` typically taking on values of 1 or 2.

Reviewer 3: ```suggestion
In the FITS standard, these keys are denoted by "CDELTi", with "i" typically taking on values of 1 or 2.
```

Mirror Reflectance
******************

The mirror reflectance, :math:`M(\lambda)`, is a dimensionless quantity describing the efficiency of the mirror(s) in the instrument as a function of wavelength.
If the instrument contains multiple mirrors (e.g. a primary and secondary mirror), this quantity is the product of the reflectance of each mirror.
:math:`M(\lambda)` should always be between 0 and 1.

Filter Transmittance
********************

The filter transmittance, :math:`F(\lambda)`, is a dimensionless quantity describing the efficiency of the filter(s) as a function of wavelength.
This is typically calculated by computing the transmittance through a given compound of a specified thickness.
In the case of a multilayer coating, the transmittance is the product of the transmittance of each layer of the coating.
Similarly, if an instrument contains multiple filters (e.g. an entrance and focal-plane filter), this quantity is the product of the transmittance of each mirror.
:math:`F(\lambda)` should always be between 0 and 1.

Quantum Efficiency
******************

The quantum efficiency, :math:`Q(\lambda)`, is a dimensionless quantity describing the efficiency of the detector.

> :math:`Q(\lambda)` should always be between 0 and 1.

Reviewer 4: There's some disagreement in the literature, but the quantum efficiency is usually the number of electron-hole pairs measured per photon. The quantity you're describing here is sometimes called the `effective` quantum efficiency.

Reviewer 4: And wikipedia calls this quantity the [external quantum efficiency](https://en.wikipedia.org/wiki/Quantum_efficiency#Types)

Author: I'm perfectly happy to rename this "effective quantum efficiency" and link to that wikipedia entry.

This quantity is independent of any instrument properties.

Wavelength Response
-------------------

The wavelength response describes the instrument sensitivity as a function of wavelength and time.
The wavelength response is defined as,

.. math::

    R(\lambda,t) = A_{\mathrm{eff}}(\lambda,t)f(\lambda)\frac{pg}{s}\quad[\mathrm{cm}^2\,\mathrm{DN}\,\mathrm{photon}^{-1}\,\mathrm{sr}\,\mathrm{pixel}^{-1}]

It has a physical type of area DN per photon solid angle per pixel.

Camera Gain
-----------

The camera gain, :math:`g`, describes the conversion between electrons and data number (DN).
This is a property of the detector.
The units of the camera gain are :math:`\mathrm{DN}\,\mathrm{electron}^{-1}`.

Photon-to-Energy Conversion
---------------------------

The photon-to-energy conversion is given by the amount of energy carried by a photon of wavelength :math:`\lambda`,

.. math::

    f(\lambda) = \frac{hc}{\lambda}\quad[\mathrm{eV}\,\mathrm{photon}^{-1}]

where :math:`h` is Planck's constant and :math:`c` is the speed of light.
It has a physical type of energy per photon.

.. note::

>     Use `~astropy.units.spectral` to provide a list of appropriate `astropy.units` equivalencies for this conversion.

Reviewer 5: ```suggestion
    Use the `~astropy.units.spectral` unit equivalency to provide a list of appropriate `astropy.units` equivalencies for this conversion.
```

Energy-to-Electron Conversion
-----------------------------

The energy-to-electron conversion, :math:`s`, describes the conversion between electrons released in the detector and the energy of an incoming photon.
This is commonly referred to as the *electron-hole-pair-creation energy*.
It has a physical type of energy per electron.
For silicon detectors, a value of :math:`s=3.65\,\mathrm{eV}\,\mathrm{electron}^{-1}` is typically used as this is approximately the energy required to free an electron in silicon.

Pixel Solid Angle
-----------------

The pixel area, :math:`p`, is the angular area in the plane of the sky subtended by a single detector pixel.
It has a physical type of solid angle per pixel.
The units of the pixel area are typically expressed as :math:`\mathrm{sr}\,\mathrm{pixel}^{-1}`.

> Typically, this quantity can be determined as the product the spatial plate scale in each direction.

Reviewer 6: ```suggestion
Typically, this quantity can be determined as the product of the spatial plate scale in each direction.
```
